tests/Integration/PrestaShopBundle/TranslationIntegrationTest.php

```diff
@@ -87,7 +87,7 @@ public function getExpectedTranslations(): iterable
             ['%message%' => 'bad idea'],
         ];
         yield 'translation with htmlspecialchars and sprintf' => [
-            '&lt;a href="test"&gt;10 Succesful deletion "&lt;b&gt;stringTest&lt;/b&gt;"&lt;/a&gt;',
+            '&lt;a href="test"&gt;10 Succesful deletion "<b>stringTest</b>"&lt;/a&gt;',
```

Comment:
htmlspecialchars should be escaping the content of the replaced token

Reply:
So we have to replace all trans tag to add 'raw' each place we want to use HTML in the BO ?

Reply:
This is already necessary in twig templates because twig escapes HTML by default

<img width="690" alt="Screenshot 2022-11-28 at 17 01 51" src="https://user-images.githubusercontent.com/1009343/204369979-71e2008a-4b90-44c4-926f-49a38b0d6305.png">

(Source: https://symfony.com/doc/4.4/translation.html#using-twig-filters)

Smarty templates seem to do this by default in [admin controllers](https://github.com/PrestaShop/PrestaShop/blob/8.0.0/config/smartyadmin.config.inc.php#L75) but apparently not on [front controllers](https://github.com/PrestaShop/PrestaShop/blob/8.0.0/config/smartyfront.config.inc.php#L160).

There's clearly some homogeneization to do.

Reply:
@eternoendless changed as discussed yesterday